sirepo/template/ml.py

```diff
@@ -620,11 +620,49 @@ def _generate_parameters_file(data):
             v,
             f"{d[dm.classificationAnimation.classifier]}.py",
         )
-    res += template_common.render_jinja(SIM_TYPE, v, "build-model.py")
+    res += _build_model_py(v)
     res += template_common.render_jinja(SIM_TYPE, v, "train.py")
     return res
 
 
+def _build_model_py(v):
+    def _import_layers(v):
+        return "".join(", " + n for n in v.layerImplementationNames if not n == "Dense")
+
+    def _layer_args(layer):
```

Comment:
I think the "d = PKDict()" is actually a constant and shouldn't get recreated on each call to _layer_args(). I'd recommend putting that as a named variable within _build_model_py() but outside of _layer_args().

Reply:
But we pass the layer into `_layer_args` each time and that will change with each call.

Reply:
Ah yes, I see. It is awkward that it constructs a full dict of values every time but really only needs one entry from it. It could be a constant PKDict of lambda functions for formatting, but the current implementation is good enough for now.

Reply:
I'll try out the lambda functions for the new layers issue. I was running into some problems and that might be a nice fix